src/GraphQL/Execution/ResolveFieldContextSlim.cs

```diff
@@ -0,0 +1,119 @@
+using GraphQL.Instrumentation;
+using GraphQL.Language.AST;
+using GraphQL.Types;
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading;
+using Field = GraphQL.Language.AST.Field;
+
+namespace GraphQL.Execution
+{
+    public static class ResolveFieldContextSlim
+    {
+        private static readonly ConcurrentDictionary<System.Type, Func<ExecutionNode, ExecutionContext, IResolveFieldContext>> dic =
+            new ConcurrentDictionary<System.Type, Func<ExecutionNode, ExecutionContext, IResolveFieldContext>>();
+
+        public static IResolveFieldContext Create(ExecutionNode node, ExecutionContext context)
+        {
+            //get the type of the source object, and just use typeof(object) when the source is null
+            var sourceType = node.Source?.GetType() ?? typeof(object);
```

Comment:
You can try to get type from resolver

Reply:
Not applicable due to performance findings

Reply:
Well, it turned out easier.